crossbar/router/broker.py

```diff
@@ -171,13 +171,17 @@ def _filter_publish_receivers(self, receivers, publish):
 
         # if "eligible_authid" we only accept receivers that have the correct authid
         if publish.eligible_authid:
+            eligible = set()
             for aid in publish.eligible_authid:
-                receivers = receivers & self._router._authid_to_sessions.get(aid, set())
```

Comment:
FWIW, I do think the wording in the WAMP spec is in line with above semantics: https://github.com/wamp-proto/wamp-proto/blob/master/rfc/text/advanced/ap_pubsub_subscriber_blackwhite_listing.md#feature-definition

But maybe the text can be improved to make it more obvious ..